### PR TITLE
Update shoc timing structure

### PR DESCRIPTION
**Pull request by tcclevenger**
Small changes to the shoc timings to avoid counting variable initialization. This makes shoc match the structure of p3.

### CODE REVIEW

**Comment by tcclevenger** [comment]
Here are some initial timings for shoc using these changes. I recorded the timings using the `./perf_analysis` scripts (11 repeats, ignore the first run, timing represents the average of the last 10 runs). I tested on Blake and Weaver. Factors are ~15x more columns/sec for CPU and ~35x more columns/sec for GPU. Are these close to expected?
Screenshot from 2021-04-12 13-08-05

File: components/eam/src/physics/cam/shoc.F90 [hunk 1]
```diff
@@ -192,7 +192,11 @@ subroutine shoc_main ( &
      w_sec, thl_sec, qw_sec, qwthl_sec,&  ! Output (diagnostic)
      wthl_sec, wqw_sec, wtke_sec,&        ! Output (diagnostic)
      uw_sec, vw_sec, w3,&                 ! Output (diagnostic)
-     wqls_sec, brunt, shoc_ql2)           ! Output (diagnostic)
+     wqls_sec, brunt, shoc_ql2,&          ! Output (diagnostic)
```
Comment (jeff-cohere) [inline components/eam/src/physics/cam/shoc.F90:195]: Does the comma at the end of this line mess things up when `SCREAM_CONFIG_IS_CMAKE` isn't defined?

File: components/scream/src/physics/shoc/shoc_main_impl.hpp [hunk 2]
```diff
@@ -384,6 +384,7 @@ Int Functions<S,D>::shoc_main(
 
     shoc_output.pblh(i)[0] = pblh_s;
   });
+  Kokkos::fence();
```
Comment (ambrad) [inline components/scream/src/physics/shoc/shoc_main_impl.hpp:387]: Nice. I was looking for this. :)

**Review by ambrad — APPROVED** [review]
Great!

For the timing stuff, I recommend a few things independent of this PR, at your convenience:
- Measure single-thread C++ vs single-thread F90 on an architecture with AVX2 or AVX512 instructions. Note that on SKX (like blake) there are tradeoffs in -xcore-avx2 vs -xcore-avx512; the latter uses wider registers but throttles the clock. So it's worth running both F90 and C++ with each of these flags.
- Test 48 threads as well as 96 on blake.
- It can be useful in the figure to give a better indication of speedup of the C++ w.r.t. to serial F90. You could added a text label to the F90 curve with the y-value for the final point, for example.
- You probably should add a C++ serial curve, too, again with a text label.
- If the parallel speedup on SKX isn't close to at least 40, I'd recommend doing an analysis to determine at least the following:
  - Is the C++ substantially slower than the F90 when run serially?
  - If so, isolate and fix hot spots.
  - If not, then sweep over thread count to see where the fall-off from perfect scaling occurs.

**Review by jeff-cohere — APPROVED** [review]

**Review by bartgol — APPROVED** [review]

**Pull request merged**